### PR TITLE
Avoid crashing on first installation

### LINKED ISSUE #20: Fix initial import errors & crash
After installing the plugin, upon initial startup, I got all of these error messages and often a crash. The plugin doesn't load until a restart. 

I think we can refactor the import files so they aren't tied to non-existent cached data.

```

ERROR: Unable to open file: res://.godot/imported/icon_terrain3d.svg-7b79ba28bfbcfe67d066affa10c60bff.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_terrain3d.svg-7b79ba28bfbcfe67d066affa10c60bff.ctex. Make sure resources have been imported by opening the project in th.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_terrain3d.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_terrain_material.svg-3e0aae3669fe2efeaf707a6892b5f063.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_terrain_material.svg-3e0aae3669fe2efeaf707a6892b5f063.ctex. Make sure resources have been imported by opening the projec.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_terrain_material.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_terrain_layer_material.svg-810362a81388cf3fc0c37a0a04912bb9.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_terrain_layer_material.svg-810362a81388cf3fc0c37a0a04912bb9.ctex. Make sure resources have been imported by opening the .
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_terrain_layer_material.svg. Make sure resources have been imported by opening the project in the editor at least on.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_map_add.svg-bc7f88fd1deaf4cbf31b1d4c8c6d0b2c.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_map_add.svg-bc7f88fd1deaf4cbf31b1d4c8c6d0b2c.ctex. Make sure resources have been imported by opening the project in the .
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_map_add.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_map_remove.svg-150dbd16519910c64a5accd6a15577ea.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_map_remove.svg-150dbd16519910c64a5accd6a15577ea.ctex. Make sure resources have been imported by opening the project in t.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_map_remove.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_height_add.svg-84485c2ba0abdc6736a7102d21854ca4.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_height_add.svg-84485c2ba0abdc6736a7102d21854ca4.ctex. Make sure resources have been imported by opening the project in t.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_height_add.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_height_sub.svg-bd2b838af515ef43c8b3c9bc412ab7b6.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_height_sub.svg-bd2b838af515ef43c8b3c9bc412ab7b6.ctex. Make sure resources have been imported by opening the project in t.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_height_sub.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_height_mul.svg-94df27cd9453a24dccf06b437c08d376.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_height_mul.svg-94df27cd9453a24dccf06b437c08d376.ctex. Make sure resources have been imported by opening the project in t.
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_height_mul.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
ERROR: Unable to open file: res://.godot/imported/icon_height_flat.svg-dcafa26c94284e57fa4f103237659863.ctex.
   at: (scene/resources/texture.cpp:814)
ERROR: Failed loading resource: res://.godot/imported/icon_height_flat.svg-dcafa26c94284e57fa4f103237659863.ctex. Make sure resources have been imported by opening the project in .
   at: (core/io/resource_loader.cpp:214)
ERROR: Failed loading resource: res://addons/terrain/icons/icon_height_flat.svg. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:214)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_terrain3d.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:4)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_TERRAIN".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:4)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_terrain_material.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:5)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_TERRAIN_MATERIAL".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:5)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_terrain_layer_material.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:6)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_LAYER_MATERIAL".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:6)
SCRIPT ERROR: Parse Error: "Terrain3D" was not found in the current scope.
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:8)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_map_add.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:183)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_MAP_ADD".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:183)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_map_remove.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:184)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_MAP_REMOVE".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:184)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_height_add.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:185)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_HEIGHT_ADD".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:185)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_height_sub.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:186)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_HEIGHT_SUB".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:186)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_height_mul.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:187)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_HEIGHT_MUL".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:187)
SCRIPT ERROR: Parse Error: Could not preload resource file "res://addons/terrain/icons/icon_height_flat.svg".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:188)
SCRIPT ERROR: Parse Error: Could not resolve type for constant "ICON_HEIGHT_FLAT".
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:188)
SCRIPT ERROR: Parse Error: Identifier "Terrain3D" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:42)
SCRIPT ERROR: Parse Error: Identifier "Terrain3D" not declared in the current scope.
          at: GDScript::reload (res://addons/terrain/editor/editor.gd:45)
```

### PR DESCRIPTION
**Pull request by lw64**
Fixes #20

### CODE REVIEW

**Comment by lw64** [comment]
Fixes #20

**Comment by lw64** [comment]
The error that makes Godot crash for me now is that it doesn't know the symbol `Terrain3dSurface` in `editor.gd` line 27:
`editor = Terrain3DEditor.new()`

1. Can Godot be made aware of this symbol somehow without restarting?
2. or how can I recognize that it the plugin is run the first time and in that case open a dialog to restart godot?
3. Would that even work because the error says it cannot "compile" the script. (Though the type name appears earlier already)

**Comment by TokisanGames** [comment]
Hmm, Maybe we need to make the change in C++, as jolt does, and catch it on the first load of the extension before any gdscript loads. 

Here's where jolt registers with Godot
https://github.com/godot-jolt/godot-jolt/blob/74ea8f3401aded9c9b833f886dbb4fae7bcfad04/src/register_types.cpp

This function calls register_settings(), which is where it initiates the restart
https://github.com/godot-jolt/godot-jolt/blob/74ea8f3401aded9c9b833f886dbb4fae7bcfad04/src/servers/jolt_project_settings.cpp#L56

This will require some exploration and experimentation with the C++. If it's getting beyond your comfort zone you could just limit this PR to the load image changes.

**Comment by lw64** [comment]
So this removes the error due to the script, but it still crashes for me. Not sure if its something completely different though:
```
WARNING: XCreateIC couldn't create wd.xic
     at: _create_window (platform/linuxbsd/x11/display_server_x11.cpp:5260)
ERROR: FATAL: Condition "!exists" is true.
   at: operator[] (./core/templates/hash_map.h:504)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v4.1.1.stable.flathub (bd6af8e0ea69167dd0627f3bd54f9105bda0f8b5)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
ERROR: FATAL: Index p_index = 1 is out of bounds (size() = 0).
   at: get (./core/templates/cowdata.h:155)
```

File: project/addons/terrain_3d/editor/editor.gd [hunk 1]
```diff
@@ -22,8 +22,14 @@ var region_gizmo: RegionGizmo
...
 var mouse_global_position: Vector3 = Vector3.ZERO
 
+var first_run: bool false
+
 
 func _enter_tree() -> void:
+	if ClassDB.class_exists("Terrain3DEditor"):
+		first_run = true
+		OS.alert("Restart the editor to use the Terrain3D plugin")
+		return
```
Comment (TokisanGames) [inline project/addons/terrain_3d/editor/editor.gd:32]: This is going to run every time, not the first time after installing the plugin only. Alright, let's drop this commit and just go with the first commit which partially resolves the issue by removing the errors. Later someone will look into initiating a restart before the crash via C++. Thanks.

**Comment by TokisanGames** [comment]
Superseded by #196

**Pull request closed without merging**